pkg/services/correlations/models.go

```diff
@@ -0,0 +1,75 @@
+package correlations
+
+import (
+	"errors"
+)
+
+var (
+	ErrSourceDataSourceReadOnly           = errors.New("source data source is read only")
+	ErrSourceDataSourceDoesNotExists      = errors.New("source data source does not exist")
+	ErrTargetDataSourceDoesNotExists      = errors.New("target data source does not exist")
+	ErrCorrelationFailedGenerateUniqueUid = errors.New("failed to generate unique correlation UID")
+	ErrCorrelationIdentifierNotSet        = errors.New("source identifier and org id are needed to be able to edit correlations")
+)
+
+// Correlation is the model for correlations definitions
+type Correlation struct {
+	ID          int64  `xorm:"pk autoincr 'id'"`
+	UID         string `xorm:"uid"`
+	SourceUID   string `xorm:"source_uid"`
```

Comment:
Given correlations are identified by their UID + the shource UID, would this make more sense instead? there's no use for the ID field

```suggestion
	UID         string `xorm:"pk 'uid'"`
	SourceUID   string `xorm:"pk 'source_uid'"`
```

Reply:
Can there be more than one correlation with a UID/SourceUID ?

Reply:
the UID is generated a creation time, and it's unique to each correlation (it's basically it's ID, having it composite with source_uid only makes having collisions harder as it then need to be unique only for that given source_uid)